File: Source/Affixes/Affix.cs
public abstract class Affix
{
    public Wolf Unit { get; set; }
    public string Name { get; set; }
    public string TypeName { get; set; }

    public Affix(Wolf unit)
    {
        Unit = unit;
        TypeName = this.GetType().Name;
    }

    public virtual void Apply()
    {
        // GC.GCAffixes.Add(this);
    }

    public virtual void Remove()
    {
        // GC.RemoveAffix(this);
    }

    public abstract void Pause(bool pause);
}

File: src/Affixes/Affix.ts
import { Wolf } from 'src/Game/Entities/Wolf'

export abstract class Affix {
    public Unit: Wolf
    public name: string = ''

    public constructor(unit: Wolf) {
        this.Unit = unit
    }

    public abstract Apply: () => void

    public abstract Remove: () => void

    public abstract pause(pause: boolean): void
}
